Tasks/ANT/ant.ps1

```diff
@@ -108,9 +108,15 @@ if ($jdkPath)
 }
 
 $buildRootPath = Split-Path $antBuildFile -Parent
-$reportDirectoryName = [guid]::NewGuid()
+$reportDirectoryName = "ReportDirectory"
 $reportDirectory = Join-Path $buildRootPath $reportDirectoryName
 
+if(Test-Path $reportDirectory)
```

Comment:
Yes, the tasks don't proceed until everything is done. It is waiting even for the async calls. So we are safe in both the scenarios you specified.